CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs

```diff
@@ -128,20 +183,65 @@ private static async Task<Operation> MonitorOperation(IQnAMakerClient qnaClient,
                 await Task.Delay(5000);
                 operation = await qnaClient.Operations.GetDetailsAsync(operation.OperationId);
             }
+            return operation;
+        }
+        // </MonitorOperation>
 
-            if (operation.OperationState != OperationStateType.Succeeded)
+        // Checks valid file type before sending for extraction
+        private static bool IsValidFile(string fileName)
+        {
+            HashSet<string> fileExtensions = new HashSet<string> { "tsv", "pdf", "txt", "docx", "xlsx" };
+            string urlExtension;
+            try
             {
-                log.LogError($"Operation {operation.OperationId} failed to completed. ErrorMessage: {operation.ErrorResponse.Error.Message}");
```

Comment:
Why did you remove this?  I would like the error message for debugging if there are errors.  This also puts it cleanly in the MonitorOperation which would apply it to all operations.

Reply:
Done in GetOperationStatus(), added comments